src/encoder.rs

```diff
@@ -91,9 +91,9 @@ where
     fn reset_buffer(&mut self) {
         // Reset buffer
         self.buffer.clear();
-        // Arbitrary bytes to take up space. Should never be seen in output.
-        let placeholder: [u8; PLACEHOLDER_HEADER_SIZE] = *b"DACE\r\n";
-        self.buffer.extend_from_slice(&placeholder);
+        // Reserve space for the chunk size. This space will be populated once
+        // we know the size of the chunk.
+        self.buffer.resize(PLACEHOLDER_HEADER_SIZE, 0);
```

Comment:
@jsha This change seem okay? We don't need to fill this allocation in with meaningful data, it's slightly more performant just to zero it out

https://godbolt.org/z/n7d7cq

https://editor.mergely.com/Z9UMLCvs/

Reply:
Ah, very nice! This works. And actually, we could do even better: instead of `buffer.clear()` above, we could do `buffer.truncate(PLACEHOLDER_HEADER_SIZE)`.

Note that if we're not longer actually sticking the placeholder in there, we could do away with the placeholder constant entirely.

Reply:
The problem with `truncate` is that it won't grow the `Vec` if the length is 0, and we need it to grow in the initialization case. I'm going to merge as is for now, but if you find a way to make `truncate` work, we can open a new PR for that!

Reply:
Nvm, we initialize the buffer vec to have 6 elements, so I think `truncate` should work

Reply:
I'm late to the party. `Vec::resize()` does a `truncate` if the desired size is smaller. Not that it matters in this specific case.